src/Grafana.OpenTelemetry.Base/Instrumentations/AWSResource.cs

```diff
@@ -0,0 +1,26 @@
+//
+// Copyright Grafana Labs
+// SPDX-License-Identifier: Apache-2.0
+//
+
+using OpenTelemetry.ResourceDetectors.AWS;
+using OpenTelemetry.Resources;
+
+namespace Grafana.OpenTelemetry
+{
+    internal class AWSResourceInitializer : InstrumentationInitializer
+    {
+        public override Instrumentation Id { get; } = Instrumentation.AWSResource;
+
+        protected override ResourceBuilder InitializeResourceDetector(ResourceBuilder builder)
+        {
+            return builder
+#if !NETFRAMEWORK && !NETSTANDARD
+                .AddDetector(new AWSECSResourceDetector())

```

Comment:
AWS ECS supports Windows nodes and the [resource detector package still has net462 support](https://www.nuget.org/packages/OpenTelemetry.ResourceDetectors.AWS/1.4.0-beta.1#supportedframeworks-body-tab).

I'm unsure if the EC2 detector would be adequate, or if we should enable ECS detection on netfx.

Reply:
@matt-hensley Could we do that just with preprocessor directives?

Reply:
I think this would cover it:
```csharp
            return builder
#if !NETSTANDARD
                .AddDetector(new AWSECSResourceDetector())
#endif
#if !NETFRAMEWORK && !NETSTANDARD
                .AddDetector(new AWSEKSResourceDetector())
#endif
                .AddDetector(new AWSEC2ResourceDetector())
                .AddDetector(new AWSEBSResourceDetector());
```

Reply:
That doesn't work, because the ECS resource detector is only available on Windows (so the above doesn't compile on Linux).

I'll merge for now, we can still iterate and refine the conditions.